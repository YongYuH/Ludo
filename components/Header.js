import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const LudoLogoSrc = '/static/LudoLogo.png';

const HeaderWrapper = styled.div`
  align-self: center;
  background-color: #717070;
  display: flex;
  height: 50px;
  position: relative;
`;

const IconWrapper = styled.div`
  align-self: center;
  display: inline-flex;
  padding: 0 20px;
`;

const LogoWrapper = IconWrapper.extend`
  align-self: center;
  margin: 0 auto;

  img {
    height: 25px;
  }
`;

const AvatarWrapper = IconWrapper.extend`
  position: absolute;
  right: 0;
`;

const PopUpMenu = styled.div`
  display: ${props => props.isShowingAvatarPopUpMenu ? 'block' : 'none'};
`;

const Logo = () => (
  <LogoWrapper>
    <img alt="logo" src={LudoLogoSrc} />
  </LogoWrapper>
);

const Menu = () => (
  <IconWrapper>
    MenuIcon
  </IconWrapper>
);

const Avatar = ({
  handleAvatarPopMenuToggle,
  isShowingAvatarPopUpMenu
}) => (
  <AvatarWrapper onClick={handleAvatarPopMenuToggle}>
    defaultAvatar
    <PopUpMenu isShowingAvatarPopUpMenu={isShowingAvatarPopUpMenu} >
      popupMenu
    </PopUpMenu>
  </AvatarWrapper>
);

Avatar.propTypes = {
  handleAvatarPopMenuToggle: PropTypes.func.isRequired,
  isShowingAvatarPopUpMenu: PropTypes.func.isRequired
};

const Header = ({
  handleAvatarPopMenuToggle,
  isShowingAvatarPopUpMenu
}) => (
  <HeaderWrapper>
    <Menu />
    <Logo />
    <Avatar
      handleAvatarPopMenuToggle={handleAvatarPopMenuToggle}
      isShowingAvatarPopUpMenu={isShowingAvatarPopUpMenu}
    />
  </HeaderWrapper>
);

Header.propTypes = {
  handleAvatarPopMenuToggle: PropTypes.func.isRequired,
  isShowingAvatarPopUpMenu: PropTypes.func.isRequired
};

export default Header;
