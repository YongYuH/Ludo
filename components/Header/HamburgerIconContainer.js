import { connect } from 'react-redux';
import HamburgerIcon from './HamburgerIcon';
import {
  getIsNavbarVisible,
  toggleNavbar,
} from '../../routes/UI/modules/navbar';

const mapStateToProps = state => ({
  isNavbarVisible: getIsNavbarVisible(state),
});

const mapDispatchToProps = dispatch => ({
  handleNavbarToggle: () => {
    dispatch(toggleNavbar());
  },
});

export default connect(mapStateToProps, mapDispatchToProps)(HamburgerIcon);
