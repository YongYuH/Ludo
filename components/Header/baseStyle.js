import styled from 'styled-components';

export const IconWrapper = styled.div`
  align-self: center;
  display: inline-flex;
  padding: 0 20px;
`;
