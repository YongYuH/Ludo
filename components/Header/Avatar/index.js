import React from 'react';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';

import * as header from '../../../redux/module/header';

import AvatarIcon from './AvatarIcon';
import AvatarPopUpMenu from './AvatarPopUpMenu';

import { IconWrapper } from '../baseStyle';

const AvatarWrapper = IconWrapper.extend`
  position: absolute;
  right: 0;
`;

export const UnconnectedAvatar = ({ toggleAvatarPopUpMenu }) => (
  <AvatarWrapper onClick={toggleAvatarPopUpMenu}>
    <AvatarIcon />
    <AvatarPopUpMenu />
  </AvatarWrapper>
);

const mapDispatchToProps = dispatch => ({
  toggleAvatarPopUpMenu: bindActionCreators(header.toggleAvatarPopUpMenu, dispatch)
});

const Avatar = connect(
  null,
  mapDispatchToProps
)(UnconnectedAvatar);

export default Avatar;
