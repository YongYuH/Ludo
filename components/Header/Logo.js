import React from 'react';
import Link from 'next/link';
import { IconWrapper } from './baseStyle';

const LudoLogoSrc = '/static/LudoLogo.png';
const LogoWrapper = IconWrapper.extend`
  img {
    width: 52px;
  }
`;

const Logo = () => (
  <LogoWrapper>
    <Link href="/index">
      <a>
        <img alt="logo" src={LudoLogoSrc} />
      </a>
    </Link>
  </LogoWrapper>
);

export default Logo;
