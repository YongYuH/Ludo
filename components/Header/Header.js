import React from 'react';
import styled from 'styled-components';

import Avatar from './Avatar/Avatar';
import Menu from './Menu/Menu';

import { IconWrapper } from './baseStyle';

const LudoLogoSrc = '/static/LudoLogo.png';

const HeaderWrapper = styled.div`
  background-color: #717070;
  display: flex;
  height: 50px;
  position: relative;
`;

const LogoWrapper = IconWrapper.extend`
  margin: 0 auto;
  padding: 0 20px;

  img {
    height: 25px;
  }
`;

const Logo = () => (
  <LogoWrapper>
    <img alt="logo" src={LudoLogoSrc} />
  </LogoWrapper>
);

const Header = () => (
  <HeaderWrapper>
    <Menu />
    <Logo />
    <Avatar />
  </HeaderWrapper>
);

export default Header;
