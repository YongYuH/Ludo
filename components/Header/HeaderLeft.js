import React from 'react';
import styled from 'styled-components';

import HamburgerIconContainer from './HamburgerIconContainer';
import Logo from './Logo';

const HeaderLeftWrapper = styled.div`
    align-items: center;
    display: inline-flex;
    height: 40px;
    justify-content: space-between;
`;

const HeaderLeft = () => (
  <HeaderLeftWrapper>
    <HamburgerIconContainer />
    <Logo />
  </HeaderLeftWrapper>
);

export default HeaderLeft;
