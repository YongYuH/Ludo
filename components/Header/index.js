import React from 'react';
import styled from 'styled-components';
import Avatar from './Avatar';
import HamburgerIcon from './HamburgerIcon';
import Logo from './Logo';

const HeaderWrapper = styled.div`
  align-items: center;
  background-color: #717070;
  display: flex;
  height: 40px;
  position: relative;
`;

const Header = () => (
  <HeaderWrapper>
    <HamburgerIcon />
    <Logo />
    <Avatar />
  </HeaderWrapper>
);

export default Header;
