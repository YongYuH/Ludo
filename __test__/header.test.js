import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';

import Header from '../components/Header/Header';

test('header style works', () => {
  const component = renderer.create(<Header />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
