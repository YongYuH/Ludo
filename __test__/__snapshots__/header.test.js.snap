// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`header style works 1`] = `
.c0 {
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  background-color: #717070;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 50px;
  position: relative;
}

.c1 {
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  padding: 0 20px;
}

.c2 {
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  padding: 0 20px;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  margin: 0 auto;
}

.c2 img {
  height: 25px;
}

.c3 {
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  padding: 0 20px;
  position: absolute;
  right: 0;
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    MenuIcon
  </div>
  <div
    className="c2"
  >
    <img
      alt="logo"
      src="/static/LudoLogo.png"
    />
  </div>
  <div
    className="c3"
  >
    PersonIcon
  </div>
</div>
`;
